src/mscorlib/src/System/Threading/Tasks/Task.cs

```diff
@@ -6603,7 +6603,7 @@ private void InvokeCoreAsync(Task completingTask)
                 // InvokeCore(completingTask);
                 var tuple = (Tuple<UnwrapPromise<TResult>, Task>)state;
                 tuple.Item1.InvokeCore(tuple.Item2);
-            }, Tuple.Create<UnwrapPromise<TResult>, Task>(this, completingTask));
+            }, Tuple.Create<UnwrapPromise<TResult>, Task>(this, completingTask), preferLocal: true);
```

Comment:
Isn't it likely this makes things worse?  As the comment indicates, we're super deep on the call stack here, and yet this change is queueing to the local queue, which means either the work item needs to wait for this thread to pick it off, which won't happen until likely thousands of stack frames are unwound, or another thread is going to incur the cost of stealing.